### PR TITLE
InternalVolume was working just fine

### PR DESCRIPTION
**Pull request by jasondhales**
If your options were chosen in line.
Input checks were inserted,
Dodgy output averted.
All this with a slight redesign.
Ref #7061

### CODE REVIEW

File: framework/src/mesh/MooseMesh.C [hunk 1]
```diff
@@ -738,7 +738,7 @@ MooseMesh::cacheInfo()
     {
       std::vector<BoundaryID> boundaryids = getBoundaryIDs(elem, side);
 
-      std::set<BoundaryID> subdomain_set = _subdomain_boundary_ids[subdomain_id];
+      std::set<BoundaryID> & subdomain_set = _subdomain_boundary_ids[subdomain_id];
```
Comment (friedmud) [inline framework/src/mesh/MooseMesh.C]: Yikes - was this just _broken_?

File: modules/misc/src/InternalVolume.C [hunk 2]
```diff
@@ -6,11 +6,13 @@
...
 #include "InternalVolume.h"
 
+#include "MooseMesh.h"
+
 template <>
 InputParameters validParams<InternalVolume>()
 {
   InputParameters params = validParams<SideIntegralPostprocessor>();
-  params.addParam<unsigned int>("component", 1, "The component to use in the integration");
+  params.addRangeCheckedParam<unsigned int>("component", 1, "component<3", "The component to use in the integration");
```
Comment (friedmud) [inline modules/misc/src/InternalVolume.C]: Does the check here need to be `component >= 0 & component < 3` ?

File: modules/misc/src/InternalVolume.C [hunk 3]
```diff
@@ -50,16 +52,68 @@ InternalVolume::InternalVolume(const InputParameters & parameters) :
...
+  {
+    const std::set<BoundaryID> & boundaries = _mesh.getSubdomainBoundaryIds(*iter);
+    std::set<BoundaryID>::const_iterator bnd = boundaries.begin();
+    const std::set<BoundaryID>::const_iterator bnd_end = boundaries.end();
+    for ( ; bnd != bnd_end; ++bnd )
+    {
+      const std::set<BoundaryID> & b = boundaryIDs();
+      std::set<BoundaryID>::const_iterator bit = b.begin();
+      const std::set<BoundaryID>::const_iterator bit_end = b.end();
+      for ( ; bit != bit_end; ++bit )
```
Comment (friedmud) [inline modules/misc/src/InternalVolume.C]: I cannot think of a better poster child for C++11 range-based-for... look at this stuff!

We're still holding off for just a little longer on allowing those... but seeing this makes me itch a little bit!

Comment (friedmud) [inline modules/misc/src/InternalVolume.C]: Seriously... 12 lines of heinous code would go down to 3 simple `for` statements.  Maybe you could get a pass from @permcody ;-)

File: modules/misc/src/InternalVolume.C [hunk 4]
```diff
@@ -50,16 +52,68 @@ InternalVolume::InternalVolume(const InputParameters & parameters) :
...
+        if (*bit == *bnd)
+        {
+          Moose::CoordinateSystemType coord_sys = _fe_problem.getCoordSystem(*iter);
+          if (_component != 0 && coord_sys == Moose::COORD_RSPHERICAL)
```
Comment (friedmud) [inline modules/misc/src/InternalVolume.C]: What if there are different regions of the domain with different coordinate systems?

I think you need to restrict this check to only the boundaries this object is restricted to.

File: modules/misc/src/InternalVolume.C (hunk 2 above)
Comment (dschwen) [inline modules/misc/src/InternalVolume.C]: There is little danger that the unsigned parameter is less than zero ;-)

File: modules/misc/src/InternalVolume.C [hunk 5]
```diff
@@ -50,16 +52,68 @@ InternalVolume::InternalVolume(const InputParameters & parameters) :
...
+    scale = 1.0;
+  }
+  return -scale*_q_point[_qp](_component)*_normals[_qp](_component);
 }
 
 Real
 InternalVolume::getValue()
 {
-  return _scale * SideIntegralPostprocessor::getValue() + _addition;
+  return _scale*SideIntegralPostprocessor::getValue()+_addition;
```
Comment (dschwen) [inline modules/misc/src/InternalVolume.C]: This makes the operator spacing worse.

File: modules/misc/src/InternalVolume.C (hunk 4 above)
Comment (jasondhales) [inline modules/misc/src/InternalVolume.C]: if (*bit == *bnd) then one of the boundaries this object is restricted to matches one of the boundaries of the block *iter.  It looks to me like the check is restricted like you mentioned.

File: modules/misc/src/InternalVolume.C (hunk 2 above)
Comment (friedmud) [inline modules/misc/src/InternalVolume.C]: I'm actually not sure of that... I don't know the specifics of when / how this check gets applied.

Plus, this is more self-documentary...

File: modules/misc/src/InternalVolume.C (hunk 4 above)
Comment (friedmud) [inline modules/misc/src/InternalVolume.C]: Ok - I see it now.  Sorry about that!  I didn't see that you were calling the local `boundaryIDs()` function for this object (I thought it was getting all of the boundary IDs in the system).

**Comment by permcody** [comment]
👍

**Pull request merged**